bin/templates/scripts/cordova/lib/projectFile.js

```diff
@@ -40,8 +40,18 @@ function parseProjectFile (locations) {
     const xcodeproj = xcode.project(pbxPath);
     xcodeproj.parseSync();
 
-    const xcBuildConfiguration = xcodeproj.pbxXCBuildConfigurationSection();
-    const plist_file_entry = _.find(xcBuildConfiguration, entry => entry.buildSettings && entry.buildSettings.INFOPLIST_FILE);
+    var projectName = '';
+    if (fs.readdirSync(project_dir).find(d => d.includes('.xcworkspace'))) {
+        projectName = fs.readdirSync(project_dir).find(d => d.includes('.xcworkspace')).replace('.xcworkspace', '');
+    }
+    var xcBuildConfiguration = xcodeproj.pbxXCBuildConfigurationSection();
+    var plist_file_entry = _.find(xcBuildConfiguration, function (entry) {
+        return (
+            entry.buildSettings &&
+            entry.buildSettings.INFOPLIST_FILE &&
+            projectName ? entry.buildSettings.INFOPLIST_FILE.includes(projectName) : true
+        );
+    });
     const plist_file = path.join(project_dir, plist_file_entry.buildSettings.INFOPLIST_FILE.replace(/^"(.*)"$/g, '$1').replace(/\\&/g, '&'));
```

Comment:
Thanks,  I have never worked in xcode before, nor in Cordova before. As far as JS goes,

`MyAppExtension-Info.plist".includes("MyApp-Info.plist")` is false. The only way to match it would be `ExtensionMyApp-Info.plist".includes("MyApp-Info.plist")`. Instead of doing the check, it might be easier to just use `entry.buildSettings.INFOPLIST_FILE === ${projectName}-Info.plist`

Reply:
Here is an actual `INFOPLIST_FILE` value that I grabbed from a new empty project.

```
"MyApp/MyApp-Info.plist"
```

Please note that the surrounding double quotes are also in string.

Reply:
Update: Here are some more sample `INFOPLIST_FILE` values. On the blank project I added some random watch extensions to create more data.

```
"MyAppWatchExtension Extension/Info.plist"
```
```
"SomeOtherWatchExtension WatchKit Extension/Info.plist"
```
```
"SomeOtherWatchExtension WatchKit App/Info.plist"
```

Not sure why this one didn't have double quotation wrapper like there is also cases where its not wrapped.
```
MyAppWatchExtension/Info.plist
```

Reply:
Also, just wanted to point out that you wrote above:

The code you showed in comment is true,
```MyAppExtension-Info.plist".includes("MyApp-Info.plist")```

But what you wrote in code didn't reflect that previously.
```
entry.buildSettings.INFOPLIST_FILE.includes(projectName)
```
for example it would have been like these which would have matched in all cases.

* `"MyApp/MyApp-Info.plist"".includes("MyApp")`
* `MyAppWatchExtension Extension/Info.plist`

Reply:
thanks